---
title: "Phgenome"
author: "Elias"
date: "July 16, 2018"
output: html_document
---
Pacific herring genome assembly and annotation
===
# Assembling genome 
* using Supernova 2.0 pipeline and 55x coverage

```{bash}
/pylon5/bi4ifup/eoziolor/program/supernova-2.0.0/supernova-cs/2.0.0/bin/run --id phgenome3 --maxreads 340000000 --fastqs /pylon5/bi4ifup/eoziolor/phgenome/data/raw/ --localcores=28
```

# Generating fasta for the genome

```{bash}
#!/bin/bash

#SBATCH -J fastagen
#SBATCH -o /home/eoziolor/phgenome/scripts/fasta/phg_fasta_%j.o
#SBATCH -e /home/eoziolor/phgenome/scripts/fasta/phg_fasta_%j.o
#SBATCH --time=7-00:00
#SBATCH --mem=60000

my_super=/home/eoziolor/program/supernova-2.0.0/supernova
my_out=/home/eoziolor/phgenome/data/assembly3/outs/assembly
my_fasta=/home/eoziolor/phgenome/data/assembly3/fasta/phgenome

$my_super mkoutput \
--asmdir=$my_out \
--outprefix=$my_fasta \
--style=megabubbles \
--headers=full
```

* I actually don't know if this is the best route for this, so I might generate a pseudohaploid fasta. I don't use the bubbles for anything in the annotation process.

## Generating pseudohaploid fasta file for further use

```{bash}
#!/bin/bash

#SBATCH -J fastagen
#SBATCH -o /home/eoziolor/phgenome/scripts/fasta/phg_fasta_%j.o
#SBATCH -e /home/eoziolor/phgenome/scripts/fasta/phg_fasta_%j.o
#SBATCH --time=0-01:00
#SBATCH --mem=60000

my_super=/home/eoziolor/program/supernova-2.0.0/supernova
my_out=/home/eoziolor/phgenome/data/assembly3/outs/assembly
my_fasta=/home/eoziolor/phgenome/data/assembly3/fasta/phgenome

$my_super mkoutput \
--asmdir=$my_out \
--outprefix=$my_fasta \
--style=pseudohap \
--headers=full
```

## Installing ARCS

```{bash}
git clone https://github.com/bcgsc/arcs.git
./autogen.sh
./configure && make
```