---
title: "Phgenome"
author: "Elias"
date: "July 16, 2018"
output: html_document
---
Pacific herring genome assembly and annotation
===
# Assembling genome 
* using Supernova 2.0 pipeline and 55x coverage

```{bash}
/pylon5/bi4ifup/eoziolor/program/supernova-2.0.0/supernova-cs/2.0.0/bin/run --id phgenome3 --maxreads 340000000 --fastqs /pylon5/bi4ifup/eoziolor/phgenome/data/raw/ --localcores=28
```

# Generating fasta for the genome

```{bash}
#!/bin/bash

#SBATCH -J fastagen
#SBATCH -o /home/eoziolor/phgenome/scripts/fasta/phg_fasta_%j.o
#SBATCH -e /home/eoziolor/phgenome/scripts/fasta/phg_fasta_%j.o
#SBATCH --time=7-00:00
#SBATCH --mem=60000

my_super=/home/eoziolor/program/supernova-2.0.0/supernova
my_out=/home/eoziolor/phgenome/data/assembly3/outs/assembly
my_fasta=/home/eoziolor/phgenome/data/assembly3/fasta/phgenome

$my_super mkoutput \
--asmdir=$my_out \
--outprefix=$my_fasta \
--style=megabubbles \
--headers=full
```

* I actually don't know if this is the best route for this, so I might generate a pseudohaploid fasta. I don't use the bubbles for anything in the annotation process.

## Generating pseudohaploid fasta file for further use

```{bash}
#!/bin/bash

#SBATCH -J fastagen
#SBATCH -o /home/eoziolor/phgenome/scripts/fasta/phg_fasta_%j.o
#SBATCH -e /home/eoziolor/phgenome/scripts/fasta/phg_fasta_%j.o
#SBATCH --time=0-01:00
#SBATCH --mem=60000

my_super=/home/eoziolor/program/supernova-2.0.0/supernova
my_out=/home/eoziolor/phgenome/data/assembly3/outs/assembly
my_fasta=/home/eoziolor/phgenome/data/assembly3/fasta/phgenome

$my_super mkoutput \
--asmdir=$my_out \
--outprefix=$my_fasta \
--style=pseudohap \
--headers=full
```

## Installing ARCS

```{bash}
git clone https://github.com/bcgsc/arcs.git
./autogen.sh
./configure && make
```

## Installing Longranger
```{bash}
curl -o longranger-2.2.2.tar.gz "http://cf.10xgenomics.com/releases/genome/longranger-2.2.2.tar.gz?Expires=1531828518&Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cDovL2NmLjEweGdlbm9taWNzLmNvbS9yZWxlYXNlcy9nZW5vbWUvbG9uZ3Jhbmdlci0yLjIuMi50YXIuZ3oiLCJDb25kaXRpb24iOnsiRGF0ZUxlc3NUaGFuIjp7IkFXUzpFcG9jaFRpbWUiOjE1MzE4Mjg1MTh9fX1dfQ__&Signature=gK9K5cKBIKxPYjcj58Z3BaRx37Hk24JU~c78wzGWOHGNQtHV3T13zeGKalGqvj0xWJoKfZf7YekPV542XrWVwITmwIzt0QLTh4yAgHqOqSL2bIMwM9SuOOK78~aJ0TQ2YGMS1Ta8v1SkJKv8BhDcaHc9YKXq~tDiJLSYnKcJ7sWTwZZLZLPW9tq3OVm4hLj27lLPkseC3LsSRHnx~6N2aFxzFeFTzFnvfWNwudGK6YRNy34hBaaI-Bn5~NoMXl2k20Cw8cBomntcg3e~C0ZrPhtQjajjYQ5Z-ws3G7SvPpWnGskCTg~xWX9D4MNdi16m~FV9WoNCeqScm8aKkU~1Hw__&Key-Pair-Id=APKAI7S6A5RYOXBWRPDA"

tar -xvzf longranger-2.2.2.tar.gzls

export PATH=/home/eoziolor/program/longranger-2.2.2:$PATH
```

## Checking MD5Sum of downloaded files

```{bash}
#!/bin/bash -l
#SBATCH -J md5check
#SBATCH -o md5sum-%j.o
#SBATCH -e md5sum-%j.o
#SBATCH -N 1
#SBATCH -n 8
#SBATCH --time=0-01:00
#SBATCH --mem=60000

md5sum /home/eoziolor/phgenome/data/raw/*.fastq.gz > /home/eoziolor/phgenome/data/raw/md5post.txt
```

* matches downloaded md5 sum

# Running ARCS pipeline

## Creating interleaved fastq

* Using proc10xG

```{bash}
git clone https://github.com/ucdavis-bioinformatics/proc10xG.git

###Creating trimmed interleaved files

#!/bin/bash -l
#SBATCH -J md5check
#SBATCH -o md5sum-%j.o
#SBATCH -e md5sum-%j.o
#SBATCH -N 1
#SBATCH -n 8
#SBATCH --time=0-02:00
#SBATCH --mem=60000

#programs and files
prog=/home/eoziolor/program/proc10xG/process_10xReads.py
file1=/home/eoziolor/phgenome/data/raw/PH-Sitka-93_S1_L008_R1_001.fastq.gz
file2=/home/eoziolor/phgenome/data/raw/PH-Sitka-93_S1_L008_R2_001.fastq.gz
output=/home/eoziolor/phgenome/data/trim/PH-Sitka-93_S1_L008_RA_001.fastq.gz

$prog \
-1 $file1 \
-2 $file2 \
-t 7 \
-o stdout | \
gzip > $output

```